network/network_test.go

```diff
@@ -306,6 +311,40 @@ func TestNewNetwork(t *testing.T) {
 	wg.Wait()
 }
 
+func TestIngressConnCount(t *testing.T) {
+	require := require.New(t)
+
+	emptyHandler := func(context.Context, message.InboundMessage) {}
+
+	_, networks, wg := newFullyConnectedTestNetwork(
+		t, []router.InboundHandler{
+			router.InboundHandlerFunc(emptyHandler),
+			router.InboundHandlerFunc(emptyHandler),
+			router.InboundHandlerFunc(emptyHandler),
+		})
+
+	require.Eventually(func() bool {
+		result := true
+		for _, net := range networks {
+			result = result && len(net.PeerInfo(nil)) == len(networks)-1
+		}
+		return result
+	}, time.Minute, time.Millisecond*10)
+
+	ingressConnections := make([]int, 0, len(networks))
+
+	for _, net := range networks {
+		ingressConnections = append(ingressConnections, net.IngressConnCount())
+	}
+
+	wg.Done()
```

Comment:
I think that the `wg.Done()` should be moved above the `Eventually` since it's a predecessor for the `Dispatch` call, which is required for the `Eventually`.

Reply:
[Moved the waitgroup](https://github.com/ava-labs/avalanchego/pull/3719/commits/f476513314ad5c85650e1558b5586ef4f80f5d48#diff-6534cc6f135999ac0a3c726bf471e812670d52b30707d8facd54fe5a61f9f58eL340)